ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ContractIdIT.scala

```diff
@@ -0,0 +1,149 @@
+import com.daml.ledger.api.refinements.ApiTypes.Party
+import com.daml.ledger.api.testtool.infrastructure.Allocation._
+import com.daml.ledger.api.testtool.infrastructure.Assertions.{assertGrpcError, fail}
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
+import com.daml.ledger.api.v1.value.{Record, RecordField, Value}
+import com.daml.ledger.client.binding.Primitive.ContractId
+import com.daml.ledger.test.semantic.ContractIdTests._
+import io.grpc.Status
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+// Check the Ledger API accepts or rejects non-suffixed contract ID.
+// - Central committer ledger implementations (sandboxes, KV...) may accept non-suffixed CID
+// - Distributed ledger implementations (e.g. Canton) must reject non-suffixed CID
+final class ContractIdIT extends LedgerTestSuite {
+
+  private[this] val nonSuffixedCid = (0 to 32).map("%02x".format(_)).mkString
+  private[this] val suffixedCid = (0 to 48).map("%02x".format(_)).mkString
+
+  private[this] def camlCase(s: String) =
+    s.split("[ -]").iterator.map(_.capitalize).mkString("")
+
+  List(
+    (nonSuffixedCid, "non-suffix", true),
+    (nonSuffixedCid, "non-suffix", false),
+    (suffixedCid, "suffix", true),
+  ).foreach { case (testedCid, cidDescription, accepted) =>
+    val result = if (accepted) "Accept" else "Reject"
+
+    def test(description: String)(
+        update: ExecutionContext => (
+            ParticipantTestContext,
+            Party,
+        ) => Future[Try[_]]
+    ) =
+      super.test(
+        result + camlCase(cidDescription) + "Cid" + camlCase(description),
+        result + "s " + cidDescription + "Contract Id in " + description,
+        allocate(SingleParty),
+      )(implicit ec => { case Participants(Participant(alpha, party)) =>
+        update(ec)(alpha, party).map {
+          case Success(_) if accepted => ()
+          case Failure(err: Throwable) if !accepted =>
+            assertGrpcError(
+              err,
+              Status.Code.INVALID_ARGUMENT,
+              s"Provided Contract ID $testedCid is not suffixed.",
+            )
+            ()
+          case otherwise =>
+            fail("Unexpected " + otherwise.fold(err => s"failure: $err", _ => "success"))
+        }
+      })
+
+    test("create argument") { implicit ec => (alpha, party) =>
+      alpha
+        .create(party, ContractRef(party, ContractId(testedCid)))
+        .transformWith(Future.successful)
+    }
+
+    test("exercise target") { implicit ec => (alpha, party) =>
+      for {
+        contractCid <- alpha.create(party, Contract(party))
+        result0 <-
+          alpha
+            .exercise(
+              party,
+              ContractId[ContractRef](testedCid).exerciseChange(_, contractCid),
+            )
+            .transformWith(Future.successful)
```

Comment:
Can you move `.transformWith(Future.successful)` to your `test` function, as you use it everywhere?

Reply:
It is not use on "exercise target".
I cannot make this test succeed with arbitrary `cid`, so I have to check it fail with "contract non found"